Gate training pybind build behind flag

JacobSzwejbka opened the pull request:
Differential Revision: D75079436

jathu commented on extension/training/CMakeLists.txt:
```diff
@@ -40,32 +40,34 @@ train_xor gflags executorch_core portable_ops_lib extension_tensor
+  if(EXECUTORCH_BUILD_XNNPACK)
+  # need to explicitly specify XNNPACK and microkernels-prod
+  # here otherwise uses XNNPACK and microkernel-prod symbols from libtorch_cpu
+  list(APPEND _pybind_training_dep_libs xnnpack_backend XNNPACK microkernels-prod)
```
nit: indent

jathu commented on scripts/build_android_library.sh:
```diff
@@ -50,6 +50,7 @@ build_android_native_library() {
     -DEXECUTORCH_BUILD_EXTENSION_MODULE=ON \
     -DEXECUTORCH_BUILD_EXTENSION_RUNNER_UTIL=ON \
     -DEXECUTORCH_BUILD_EXTENSION_TENSOR=ON \
+    -DEXECUTORCH_BUILD_EXTENSION_TRAINING=ON \
```
So why are we turning this on by default now?

JacobSzwejbka replied:
@kirklandsign  and @larryliu0820  recommended just piggybacking on top of the existing android ci to get build coverage.

JacobSzwejbka committed:
Gate training pybind build behind flag (#11001)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/11001

Differential Revision: D75079436
